src/sync/sync_config.hpp

```diff
@@ -115,10 +115,11 @@ struct SyncConfig {
     bool client_validate_ssl = true;
     util::Optional<std::string> ssl_trust_certificate_path;
     std::function<sync::Session::SSLVerifyCallback> ssl_verify_callback;
-    bool validate_sync_history = true;
     bool is_partial = false;
     util::Optional<std::string> custom_partial_sync_identifier;
 
+    bool validate_sync_history = true;
+
```

Comment:
As a short term solution, I would suggest committing to having the constructor. This introduces a fortunate decoupling from the actual member layout.

As a longer term solution, I propose that user sites declare default instances of `SyncConfig`, and then assign explicitly to its members.

Anyway, I agree that your fix works.